src/workarounds/gpu_arrays.jl

```diff
@@ -17,3 +17,11 @@ function LinearAlgebra.eigen(A::Hermitian{T,AT}) where {T <: Real,AT <: CuArray}
     vals, vects = CUDA.CUSOLVER.syevd!('V','U', A.data)
     (vectors = vects, values = vals)
 end
+
+# Create an array of same type as X filled with zeros, minimizing the number
+# of allocations.
+function zeros_like(X, n, m)
+    Z = similar(X, n, m)
+    Z .= 0
+    Z
+end
```

Comment:
Not really GPU-specific. I'd move it to other.

But to be completely honest: I don't think we even need it at all, because
`Z = similar(X, n, m) .= 0`
Does exactly the same thing, is more general (as other forms of similar could be used) and is also just one line.

@antoine-levitt Don't you think that's sufficient for now?

Reply:
The reason I put it in the GPU workarounds is that I had to use `zeros(similar())` to distinguish between the cases where `X` is a GPU array and `X` a CPU array. For the CPU version, there is no need for this, as we can directly call `zeros`: but when mixing CPU and GPU, what we really want to do is call Base.zeros, or CUDA.zeros (or AMDGPU.zeros...) depending on the input type of `X`. The only way I found to have this type of behavior is to first use similar.

Reply:
> But to be completely honest: I don't think we even need it at all, because Z = similar(X, n, m) .= 0 does exactly the same thing, is more general (as other forms of similar could be used) and is also just one line.

This is a bit ugly, and also it's possible that this can be optimized in the future (eg for arrays `zeros` does exactly this right now, but could conceivably use `calloc` in future versions of julia), so having a function in src/common makes it easier. You can have this version in src/common, then overload for X::CuArray in the workarounds if you want

Reply:
ok, let's have a `zeros_like` which works like `similar` but allocates zeros and put it in common. If we want to avoid direct dependency on CUDA (which we should!) we have no other option than avoiding the use of CUDA.zeros. But of course we could still specialise for `<: Array` arguments.